### PR TITLE
[CUDA] wire up cooperative kernel launch

### PR DESCRIPTION
**Pull request by JackAKirk**
The DPC++ `root_group` test already passes (and still does with this patch), but it doesn't actually enable `root_group` sync in reality. However I think the only thing missing is this small patch. With this patch the the custom kernel launch is called with the cooperative attribute. 

I've tested this locally, but will link up a dpc++ ci test for completeness.

### CODE REVIEW

**Review by konradkusiak97 — APPROVED** [review]
LGTM if all tests pass

**Comment by JackAKirk** [comment]
dpc++ test CI here: https://github.com/intel/llvm/pull/13933

**Comment by JackAKirk** [comment]
Marking it as ready to merge again. The query is blocked by https://github.com/oneapi-src/unified-runtime/issues/1687
We may as well merge this part I think.

**Pull request merged**
